service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/backfill/RetrieveControlledBigQueryDatasetsLifetimeStep.java

```diff
@@ -0,0 +1,141 @@
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_TABLE_LIFETIME_MAP;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CONTROLLED_RESOURCE_ID_TO_WORKSPACE_ID_MAP;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.cloud.gcp.bqdataset.ControlledBigQueryDatasetResource;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.api.services.bigquery.model.Dataset;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import kotlin.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO (PF-2269): Clean this up once the back-fill is done in all Terra environments.
+public class RetrieveControlledBigQueryDatasetsLifetimeStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(RetrieveControlledBigQueryDatasetsLifetimeStep.class);
+  private final CrlService crlService;
+
+  public RetrieveControlledBigQueryDatasetsLifetimeStep(CrlService crlService) {
+    this.crlService = crlService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    validateRequiredEntries(
+        context.getWorkingMap(), CONTROLLED_BIG_QUERY_DATASETS_WITHOUT_LIFETIME);
+    List<ControlledBigQueryDatasetResource> controlledBigQueryDatasets =
+        Preconditions.checkNotNull(
+            context
+                .getWorkingMap()
+                .get(CONTROLLED_BIG_QUERY_DATASETS_WITHOUT_LIFETIME, new TypeReference<>() {}));
+    Map<UUID, String> resourceIdToDefaultTableLifetimeMap = new HashMap<>();
+    Map<UUID, String> resourceIdToDefaultPartitionLifetimeMap = new HashMap<>();
+    Map<UUID, String> resourceIdToWorkspaceIdMap = new HashMap<>();
+
+    for (var resource : controlledBigQueryDatasets) {
+      logger.info(
+          "Getting default table lifetime and partition life for resource (BigQuery dataset) {} in workspace {}",
+          resource.getResourceId(),
+          resource.getWorkspaceId());
+
+      populateMapsWithResourceIdKey(
+          resourceIdToDefaultTableLifetimeMap,
+          resourceIdToDefaultPartitionLifetimeMap,
+          resourceIdToWorkspaceIdMap,
+          resource,
+          getBqDatasetDefaultTableLifetimeAndPartitionLifetime(resource));
```

Comment:
Nit: Why pass this as a parameter? I think it would be clearer if you called it inside the populate code.